### PR TITLE
EL-3382 - Fixing Protractor Tests

### PR DESCRIPTION
**Pull request by ashley-hunter**
- Our selection tests were failing - turns out the event was being emitted repeatedly, so I have now added a check to only emit when it has actually changed
- Hierarchy bar had recently changed, so I have now updated the tests to reflect the new changes.

#### Ticket
https://autjira.microfocus.com/browse/EL-3392

#### Documentation CI URL
https://pages.github.houston.softwaregrp.net/sepg-docs-qa/UXAspects_CI_UXAspects_EL-3382-SEPG-Automated-Tests

### CODE REVIEW

File: e2e/tests/components/hierarchy-bar/hierarchy-bar.e2e-spec.ts [hunk 1]
```diff
@@ -102,7 +102,7 @@ describe('Hierarchy Bar Tests', () => {
         const titles = await page.getOverflowNodeTitles();
 
         // expect there to be nodes in the overflow popover
-        expect(JSON.stringify(titles)).toBe(JSON.stringify(['Theresa Chandler']));
+        expect(titles).toEqual(['Theresa Chandler', 'Lilly Shaw']);
```
[the remark this answers is not in the document]
Comment (donna-marijne) [reply e2e/tests/components/hierarchy-bar/hierarchy-bar.e2e-spec.ts]: Do you know why it wouldn't be reliable? Are the changes the window dimensions still required for this test?

[the remark this answers is not in the document]
Comment (ashley-hunter) [reply e2e/tests/components/hierarchy-bar/hierarchy-bar.e2e-spec.ts]: Yes, we need to make the window smaller to get items overflowing. 

If I run the tests (not in headless mode) and put a breakpoint in the test after the window resizes there are two items overflowing for me:

image

**Comment by ashley-hunter** [comment]
I have commented out the test that is not working and updated the pom.xml file

**Comment by ashley-hunter** [comment]
Tests are failing because it is not using the latest changes in the buildenv-image repo. How do we update the image being used?

**Comment by donna-marijne** [comment]
You'll need to bump the version number if needed, update the [release-notes file](https://github.com/UXAspects/buildenv-image/blob/master/release-notes-1.1.0.md) to remove "!not-ready-for-release!", make a [release candidate build](https://jenkins.hpeswlab.net/job/SEPG/job/UXAspects/view/Builds/job/UXAspects~buildenv-image~master/), and then run the auto-release task with the resulting release token (found in the build log).

**Comment by ashley-hunter** [comment]
Can I just clarify, updating the version number, is the version number in the pom.xml? Eg:

```
<version>1.1.0-SNAPSHOT</version>
```
So should that be changed to like 1.2.0-SNAPSHOT or 1.1.1-SNAPSHOT?

Does the readme file name need updated to reflect the new version number as well?

**Comment by donna-marijne** [comment]
I just checked, and the version number has already been incremented, so you can ignore that part. You can verify by checking the [official-build.props](https://github.com/UXAspects/UXAspects/blob/develop/official-build.props) file and noticing that it refers to version 1.0.0.

````
SEPG_BUILD_ENV_IMAGE=uxaspects/buildenv:1.0.0
````

So you should just be able to remove the !not-ready-for-release! line and then make a build.

After making the release, bump the version in official-build.props (this only applies to the UXAspects repo at present, we are using a different image for the MF repo).

**Comment by ashley-hunter** [comment]
Ah ok, thanks very much for clarifying! I do that now

**Comment by donna-marijne** [comment]
2 failed tests in the latest SEPG build:

````
[13:16:54] I/testLogger - [chrome #01-2] PID: 484
[chrome #01-2] Specs: /wd/e2e/tests/components/column-resizing/column-resizing.e2e-spec.ts
[chrome #01-2] 
[chrome #01-2] [13:16:25] I/direct - Using ChromeDriver directly...
[chrome #01-2] AppDir: ./
[chrome #01-2]   Suite: Column Resizing Tests
[chrome #01-2]     passed - should have correct initial states (standard table)
[chrome #01-2]     passed - should have correct initial states (fixed table)
[chrome #01-2]     passed - can increase the width of a column (standard table)
[chrome #01-2]     failed - can increase the width of a column (fixed table)
[chrome #01-2]     Expected 365 to be 366.
[chrome #01-2]     Expected 195 to be 194.
[chrome #01-2]     passed - can decrease the width of a column (standard table)
[chrome #01-2]     failed - can decrease the width of a column (fixed table)
[chrome #01-2]     Expected 165 to be 166.
[chrome #01-2]     Expected 395 to be 394.
[chrome #01-2]     passed - can force update layout after pagination (fixed table)
[chrome #01-2]   Suite failed: Column Resizing Tests
[chrome #01-2] chrome
````

**Comment by ashley-hunter** [comment]
Thanks, the flex table column sizes can be a decimal value so rounding can occur. I have updated the tests to allow for a small amount of variation in column size based on rounding

**Comment by ashley-hunter** [comment]
The build seems to have run successfully and the tests passed. The only caveat is that there seems to be some issue with the `makeReport` grunt task, which throws an error `Fatal error: ENOENT: no such file or directory, open '/wd/e2e/pages/app/app.component.js'` which doesn't provide much detail about what is causing the issue.

I have removed the makeReport task and it all works as expected. Would it be possible to raise a separate ticket to re-add the makeReport ticket, as I am not able to reproduce this issue locally and I doubt I will be able to get a solution before the sprint ends?

**Pull request merged**